app/models/talentlms/facade.rb

```diff
@@ -89,14 +88,31 @@ def course_url(user, course_id, redirect_url, logout_url)
       encoded_redirect_url = Base64.strict_encode64(redirect_url)
       encoded_logout_url = Base64.strict_encode64(logout_url)
       result = @api.get('gotocourse',
-        {
-          course_id: course_id,
-          user_id: login.lms_user_id,
-          course_completed_redirect: encoded_redirect_url,
-          logout_redirect: encoded_logout_url,
-        },
-      )
+                        {
+                          course_id: course_id,
+                          user_id: login.lms_user_id,
+                          course_completed_redirect: encoded_redirect_url,
+                          logout_redirect: encoded_logout_url,
+                        })
       result['goto_url']
     end
+
+    # Generate random password for talentlms user creation
+    #
+    # An ArgumentError is raised if the length is less than 8
+    #
+    # @param length [Integer] number of characters to generate
+    # @return [String] randomly generated string of the requested length with at least on upper, lower, numeric, and symbol character
+    def random_password(length = 16)
+      raise ArgumentError 'Length must be at least 8' if length < 8
+
+      p = SecureRandom.urlsafe_base64(length - 4)
+      lower_letter = ('a'..'z').to_a.sample
+      upper_letter = ('A'..'Z').to_a.sample
+      number = ('0'..'9').to_a.sample
+      special = (('#'..'&').to_a + (':'..'?').to_a).sample
+
+      "#{p}#{lower_letter}#{upper_letter}#{number}#{special}".chars.shuffle.join
```

Comment:
Ah, good catch!  After talking about this, I was thinking 
```
[
  p,
  lower_letter,
  upper_letter,
  number,
  special,
].shuffle.join
```
would be another approach, but I think I like yours better.  Thanks!